src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -107,6 +130,8 @@
         this.clickAction = Optional.<ClickAction<?>>ofNullable(clickAction);
         this.hoverAction = Optional.<HoverAction<?>>ofNullable(hoverAction);
         this.shiftClickAction = Optional.<ShiftClickAction<?>>ofNullable(shiftClickAction);
+        this.childrenIterable = () ->
+                this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
```

Comment:
That would only work in the constructor, because the children text wouldn't be set at the point the field initializers are getting called.